sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -778,5 +778,11 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def clear(): Unit = {
     settings.clear()
   }
+
+  def reset(): Unit = {
+    sqlConfEntries.values.asScala.foreach { entry =>
```

Comment:
The major difference is our target is not to remove the `sqlConfEntries`, but to remove the entries in `settings` that stores mutable config parameters. Thanks!